test/attr/attr_availability_maccatalyst.swift

```diff
@@ -0,0 +1,125 @@
+// RUN: %swift -typecheck -verify -parse-stdlib -target x86_64-apple-ios51.0-macabi %s
+
+// REQUIRES: OS=maccatalyst
+
+@available(macCatalyst, introduced: 1.0, deprecated: 2.0, obsoleted: 9.0,
+           message: "you don't want to do that anyway")
+func obsoletedOnMacCatalyst() { }
+// expected-note @-1{{'obsoletedOnMacCatalyst()' was obsoleted in Mac Catalyst 9.0}}
+
+obsoletedOnMacCatalyst() // expected-error{{'obsoletedOnMacCatalyst()' is unavailable in Mac Catalyst: you don't want to do that anyway}}
+
+@available(iOS, introduced: 1.0, deprecated: 2.0, obsoleted: 9.0,
+           message: "you don't want to do that anyway")
+func obsoletedOnIOS() { }
+// expected-note @-1{{'obsoletedOnIOS()' was obsoleted in iOS 9.0}}
+
+obsoletedOnIOS() // expected-error{{'obsoletedOnIOS()' is unavailable in iOS: you don't want to do that anyway}}
+
+
+@available(iOS, introduced: 1.0)
+@available(macCatalyst, introduced: 1.0, obsoleted: 12.0)
+func obsoletedOnMacCatalystButNotIOS() { }
+// expected-note @-1{{'obsoletedOnMacCatalystButNotIOS()' was obsoleted in Mac Catalyst 12.0}}
+
+obsoletedOnMacCatalystButNotIOS() // expected-error {{'obsoletedOnMacCatalystButNotIOS()' is unavailable}}
+
+
+@available(iOS, introduced: 12.0, obsoleted: 12.0)
+@available(macCatalyst, introduced: 12.0)
+func obsoletedOnIOSButNotMacCatalyst() { }
+obsoletedOnIOSButNotMacCatalyst() // no-error
+
+
+
+@available(iOS, introduced: 1.0)
+@available(macCatalyst, introduced: 1.0, deprecated: 12.0)
+func deprecatedOnMacCatalystButNotIOS() { }
+
+deprecatedOnMacCatalystButNotIOS() // expected-warning {{deprecatedOnMacCatalystButNotIOS()' was deprecated in Mac Catalyst 12.0}}
+
+@available(iOS, introduced: 12.0, deprecated: 13.0)
+@available(macCatalyst, introduced: 12.0)
+func deprecatedOnIOSButNotMacCatalyst() { }
+deprecatedOnIOSButNotMacCatalyst() // no-warning
+
+
+@available(iOS 55.0, macCatalyst 56.0, *)
+func introducedLaterOnmacCatalyst() {
```

Comment:
Nit: This test function's name is awkwardly capitalized.

Reply:
Updated.